code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm

```diff
@@ -0,0 +1,103 @@
+// April 3rd, 2014 marks the day this machine changed the face of the kitchen on NTStation13
+// God bless America.
+/obj/machinery/deepfryer
+	name = "deep fryer"
+	desc = "Deep fried <i>everything</i>."
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "fryer_off"
+	layer = 2.9
+	density = 1
+	anchored = 1
+	use_power = 1
+	idle_power_usage = 5
+	flags = OPENCONTAINER
+	var/obj/item/frying = null	//What's being fried RIGHT NOW?
+	var/cook_time = 0
+
+/obj/machinery/deepfryer/New()
+	..()
+	create_reagents(50)
+	reagents.add_reagent("nutriment", 25)
+	component_parts = list()
+	component_parts += new /obj/item/weapon/circuitboard/machine/deep_fryer(null)
+	component_parts += new /obj/item/weapon/stock_parts/micro_laser(null)
+	RefreshParts()
+
+/obj/machinery/deepfryer/examine()
+	..()
+	if(frying)
+		usr << "You can make out [frying] in the oil."
+
+/obj/machinery/deepfryer/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/weapon/reagent_containers/food/snacks/deepfryholder))
+		user << "<span class='userdanger'>Your cooking skills are not up to the legendary Doublefry technique.</span>"
+		return
+	else
+		if(user.drop_item() && !frying)
+			user << "<span class='notice'>You put [I] into [src].</span>"
+			frying = I
+			frying.loc = src
+			icon_state = "fryer_on"
+
+/obj/machinery/deepfryer/process()
+	..()
+	if (!reagents.has_reagent("nutriment"))
+		reagents.add_reagent("nutriment", 25)
+	if(frying)
+		cook_time++
+		if(cook_time == 30)
+			playsound(src.loc, "sound/machines/ding.ogg", 50, 1)
+			visible_message("[src] dings!")
+		else if (cook_time == 60)
+			visible_message("[src] emits an acrid smell!")
+
+
+/obj/machinery/deepfryer/attack_hand(mob/user)
+	if(frying)
+		user << "<span class='notice'>You eject [frying] from [src].</span>"
+		if(frying.loc == src)
+			var/obj/item/weapon/reagent_containers/food/snacks/deepfryholder/S = new(get_turf(src))
+			if(istype(frying, /obj/item/weapon/reagent_containers/))
+				var/obj/item/weapon/reagent_containers/food = frying
+				food.reagents.trans_to(S, food.reagents.total_volume)
+			S.icon = frying.icon
+			S.overlays = frying.overlays
+			S.icon_state = frying.icon_state
+			S.desc = frying.desc
+			src.reagents.trans_to(S, 2*cook_time)
```

Comment:
I'm not worried about balancing the chef.  I'm worried about people taking one bite out of it and turning morbidly obese.